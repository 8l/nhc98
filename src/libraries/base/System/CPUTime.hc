#include "newmacros.h"
#include "runtime.h"

#define FN_System_46CPUTime_46realToInteger	((void*)startLabel+12)
#define CT_v186	((void*)startLabel+36)
#define F0_System_46CPUTime_46realToInteger	((void*)startLabel+44)
extern Node FN_Prelude_46realToFrac[];
extern Node CF_Prelude_46Fractional_46Prelude_46Double[];
extern Node FN_NHC_46Internal_46_95apply1[];
extern Node FN_Prelude_46RealFrac_46Prelude_46Double_46round[];
extern Node CF_Prelude_46Integral_46Prelude_46Integer[];

static Node startLabel[] = {
  bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v186)
,	/* FN_System_46CPUTime_46realToInteger: (byte 0) */
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_I3,HEAP_ARG,1)
, bytes2word(HEAP_CVAL_I4,HEAP_CVAL_I5,HEAP_OFF_N1,4)
, bytes2word(HEAP_ARG,2,PUSH_HEAP,HEAP_CVAL_P1)
, bytes2word(6,HEAP_CVAL_P1,7,HEAP_OFF_N1)
, bytes2word(5,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
,	/* CT_v186: (byte 0) */
  HW(5,2)
, 0
,	/* F0_System_46CPUTime_46realToInteger: (byte 0) */
  CAPTAG(useLabel(FN_System_46CPUTime_46realToInteger),2)
, VAPTAG(useLabel(FN_Prelude_46realToFrac))
, useLabel(CF_Prelude_46Fractional_46Prelude_46Double)
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46RealFrac_46Prelude_46Double_46round))
, useLabel(CF_Prelude_46Integral_46Prelude_46Integer)
,};
