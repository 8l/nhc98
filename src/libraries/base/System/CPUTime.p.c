#include "newmacros.h"
#include "runtime.h"

#define FN_System_46CPUTime_46realToInteger	((void*)startLabel+12)
#define CT_v191	((void*)startLabel+128)
#define F0_System_46CPUTime_46realToInteger	((void*)startLabel+136)
#define ST_v185	((void*)startLabel+176)
#define PP_System_46CPUTime_46realToInteger	((void*)startLabel+191)
#define PC_System_46CPUTime_46realToInteger	((void*)startLabel+191)
#define ST_v187	((void*)startLabel+191)
#define PS_v189	((void*)startLabel+220)
#define PS_v188	((void*)startLabel+232)
#define PS_v186	((void*)startLabel+244)
#define PS_v190	((void*)startLabel+256)
extern Node FN_Prelude_46realToFrac[];
extern Node CF_Prelude_46Fractional_46Prelude_46Double[];
extern Node FN_NHC_46Internal_46_95apply1[];
extern Node FN_Prelude_46RealFrac_46Prelude_46Double_46round[];
extern Node CF_Prelude_46Integral_46Prelude_46Integer[];
extern Node PC_NHC_46Internal_46_95apply1[];
extern Node PC_Prelude_46realToFrac[];
extern Node PC_Prelude_46RealFrac_46Prelude_46Double_46round[];

static Node startLabel[] = {
  bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v191)
,	/* FN_System_46CPUTime_46realToInteger: (byte 0) */
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_P1,7,HEAP_CVAL_N1)
, bytes2word(7,HEAP_CREATE,HEAP_SPACE,HEAP_SPACE)
, bytes2word(HEAP_ARG,1,HEAP_CVAL_P1,8)
, bytes2word(HEAP_CVAL_P1,9,HEAP_CVAL_N1,12)
, bytes2word(HEAP_CREATE,HEAP_SPACE,HEAP_SPACE,HEAP_OFF_N1)
, bytes2word(12,HEAP_ARG,2,PUSH_HEAP)
, bytes2word(HEAP_CVAL_P1,10,HEAP_CVAL_N1,17)
, bytes2word(HEAP_CREATE,HEAP_SPACE,HEAP_SPACE,HEAP_CVAL_P1)
, bytes2word(11,HEAP_OFF_N1,13,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 0
, useLabel(PS_v190)
, 0
, 0
, 0
, 0
, useLabel(PS_v189)
, 0
, 0
, 0
, 0
, useLabel(PS_v188)
, 0
, 0
, 0
, 0
, 420001
, useLabel(ST_v187)
,	/* CT_v191: (byte 0) */
  HW(5,2)
, 0
,	/* F0_System_46CPUTime_46realToInteger: (byte 0) */
  CAPTAG(useLabel(FN_System_46CPUTime_46realToInteger),2)
, useLabel(PS_v186)
, 0
, 0
, 0
, VAPTAG(useLabel(FN_Prelude_46realToFrac))
, useLabel(CF_Prelude_46Fractional_46Prelude_46Double)
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46RealFrac_46Prelude_46Double_46round))
, useLabel(CF_Prelude_46Integral_46Prelude_46Integer)
,};
Node PM_System_46CPUTime[] = {
 	/* ST_v185: (byte 0) */
  bytes2word(83,121,115,116)
, bytes2word(101,109,46,67)
, bytes2word(80,85,84,105)
,	/* PP_System_46CPUTime_46realToInteger: (byte 3) */
 	/* PC_System_46CPUTime_46realToInteger: (byte 3) */
 	/* ST_v187: (byte 3) */
  bytes2word(109,101,0,83)
, bytes2word(121,115,116,101)
, bytes2word(109,46,67,80)
, bytes2word(85,84,105,109)
, bytes2word(101,46,114,101)
, bytes2word(97,108,84,111)
, bytes2word(73,110,116,101)
, bytes2word(103,101,114,0)
,	/* PS_v189: (byte 0) */
  useLabel(PM_System_46CPUTime)
, useLabel(PP_System_46CPUTime_46realToInteger)
, useLabel(PC_NHC_46Internal_46_95apply1)
,	/* PS_v188: (byte 0) */
  useLabel(PM_System_46CPUTime)
, useLabel(PP_System_46CPUTime_46realToInteger)
, useLabel(PC_Prelude_46realToFrac)
,	/* PS_v186: (byte 0) */
  useLabel(PM_System_46CPUTime)
, useLabel(PP_System_46CPUTime_46realToInteger)
, useLabel(PC_System_46CPUTime_46realToInteger)
,	/* PS_v190: (byte 0) */
  useLabel(PM_System_46CPUTime)
, useLabel(PP_System_46CPUTime_46realToInteger)
, useLabel(PC_Prelude_46RealFrac_46Prelude_46Double_46round)
,};
