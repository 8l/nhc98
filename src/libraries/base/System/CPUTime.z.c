#include "newmacros.h"
#include "runtime.h"

#define FN_System_46CPUTime_46realToInteger	((void*)startLabel+12)
#define CT_v187	((void*)startLabel+48)
#define F0_System_46CPUTime_46realToInteger	((void*)startLabel+56)
#define ST_v185	((void*)startLabel+80)
#define ST_v186	((void*)startLabel+95)
extern Node TM_System_46CPUTime[];
extern Node FN_Prelude_46realToFrac[];
extern Node CF_Prelude_46Fractional_46Prelude_46Double[];
extern Node FN_NHC_46Internal_46_95apply1[];
extern Node FN_Prelude_46RealFrac_46Prelude_46Double_46round[];
extern Node CF_Prelude_46Integral_46Prelude_46Integer[];

static Node startLabel[] = {
  bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v187)
,	/* FN_System_46CPUTime_46realToInteger: (byte 0) */
  useLabel(TM_System_46CPUTime)
, bytes2word(NEEDHEAP_I32,HEAP_CVAL_I3,HEAP_ARG,1)
, bytes2word(HEAP_CVAL_I4,HEAP_CVAL_I5,HEAP_OFF_N1,4)
, bytes2word(HEAP_ARG,2,PUSH_HEAP,HEAP_CVAL_P1)
, bytes2word(6,HEAP_CVAL_P1,7,HEAP_OFF_N1)
, bytes2word(5,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
, 420001
, useLabel(ST_v186)
,	/* CT_v187: (byte 0) */
  HW(5,2)
, 0
,	/* F0_System_46CPUTime_46realToInteger: (byte 0) */
  CAPTAG(useLabel(FN_System_46CPUTime_46realToInteger),2)
, VAPTAG(useLabel(FN_Prelude_46realToFrac))
, useLabel(CF_Prelude_46Fractional_46Prelude_46Double)
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46RealFrac_46Prelude_46Double_46round))
, useLabel(CF_Prelude_46Integral_46Prelude_46Integer)
,};
Node PM_System_46CPUTime[] = {
 	/* ST_v185: (byte 0) */
  bytes2word(83,121,115,116)
, bytes2word(101,109,46,67)
, bytes2word(80,85,84,105)
,	/* ST_v186: (byte 3) */
  bytes2word(109,101,0,83)
, bytes2word(121,115,116,101)
, bytes2word(109,46,67,80)
, bytes2word(85,84,105,109)
, bytes2word(101,46,114,101)
, bytes2word(97,108,84,111)
, bytes2word(73,110,116,101)
, bytes2word(103,101,114,0)
,};
