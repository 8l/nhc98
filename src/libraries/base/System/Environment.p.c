#include "newmacros.h"
#include "runtime.h"

#define ST_v158	((void*)startLabel+0)

static Node startLabel[] = {
 };
Node PM_System_46Environment[] = {
 	/* ST_v158: (byte 0) */
  bytes2word(83,121,115,116)
, bytes2word(101,109,46,69)
, bytes2word(110,118,105,114)
, bytes2word(111,110,109,101)
, bytes2word(110,116,0,0)
,};
