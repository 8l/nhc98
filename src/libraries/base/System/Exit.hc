#include "newmacros.h"
#include "runtime.h"

#define CT_v172	((void*)startLabel+32)
#define CT_v173	((void*)startLabel+80)
extern Node FN_System_46exitWith[];

static Node startLabel[] = {
  bytes2word(0,0,0,0)
, useLabel(CT_v172)
,};
Node FN_System_46Exit_46exitSuccess[] = {
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_N1,1,PUSH_HEAP)
, bytes2word(HEAP_CVAL_I3,HEAP_OFF_N1,2,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 0
, CONSTR(0,0,0)
,	/* CT_v172: (byte 0) */
  HW(1,0)
, 0
,};
Node CF_System_46Exit_46exitSuccess[] = {
  VAPTAG(useLabel(FN_System_46Exit_46exitSuccess))
, VAPTAG(useLabel(FN_System_46exitWith))
, bytes2word(0,0,0,0)
, useLabel(CT_v173)
,};
Node FN_System_46Exit_46exitFailure[] = {
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_N1,1,HEAP_INT_P1)
, bytes2word(1,PUSH_HEAP,HEAP_CVAL_I3,HEAP_OFF_N1)
, bytes2word(3,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
, 0
, CONSTR(1,1,0)
,	/* CT_v173: (byte 0) */
  HW(1,0)
, 0
,};
Node CF_System_46Exit_46exitFailure[] = {
  VAPTAG(useLabel(FN_System_46Exit_46exitFailure))
, VAPTAG(useLabel(FN_System_46exitWith))
,};
