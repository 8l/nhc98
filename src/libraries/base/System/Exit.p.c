#include "newmacros.h"
#include "runtime.h"

#define CT_v176	((void*)startLabel+104)
#define CT_v181	((void*)startLabel+244)
#define ST_v171	((void*)startLabel+276)
#define ST_v178	((void*)startLabel+288)
#define ST_v173	((void*)startLabel+312)
#define PS_v180	((void*)startLabel+336)
#define PS_v179	((void*)startLabel+348)
#define PS_v177	((void*)startLabel+360)
#define PS_v175	((void*)startLabel+372)
#define PS_v174	((void*)startLabel+384)
#define PS_v172	((void*)startLabel+396)
extern Node FN_System_46exitWith[];
extern Node PC_System_46exitWith[];
extern Node PC_System_46ExitFailure[];
extern Node PC_System_46ExitSuccess[];

static Node startLabel[] = {
  bytes2word(0,0,0,0)
, useLabel(CT_v176)
,};
Node FN_System_46Exit_46exitSuccess[] = {
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_N1,7,HEAP_CVAL_N1)
, bytes2word(12,HEAP_CREATE,HEAP_SPACE,HEAP_SPACE)
, bytes2word(PUSH_HEAP,HEAP_CVAL_P1,7,HEAP_CVAL_N1)
, bytes2word(17,HEAP_CREATE,HEAP_SPACE,HEAP_SPACE)
, bytes2word(HEAP_OFF_N1,10,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
, 0
, useLabel(PS_v175)
, 0
, 0
, 0
, 0
, useLabel(PS_v174)
, 0
, 0
, 0
, 0
, CONSTR(0,0,0)
, 0
, 0
, 0
, 0
, 910001
, useLabel(ST_v173)
,	/* CT_v176: (byte 0) */
  HW(1,0)
, 0
,};
Node CF_System_46Exit_46exitSuccess[] = {
  VAPTAG(useLabel(FN_System_46Exit_46exitSuccess))
, useLabel(PS_v172)
, 0
, 0
, 0
, VAPTAG(useLabel(FN_System_46exitWith))
, bytes2word(0,0,0,0)
, useLabel(CT_v181)
,};
Node FN_System_46Exit_46exitFailure[] = {
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_N1,7,HEAP_CVAL_N1)
, bytes2word(12,HEAP_CREATE,HEAP_SPACE,HEAP_SPACE)
, bytes2word(HEAP_INT_P1,1,PUSH_HEAP,HEAP_CVAL_P1)
, bytes2word(7,HEAP_CVAL_N1,17,HEAP_CREATE)
, bytes2word(HEAP_SPACE,HEAP_SPACE,HEAP_OFF_N1,11)
, bytes2word(RETURN_EVAL,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, 0
, useLabel(PS_v180)
, 0
, 0
, 0
, 0
, useLabel(PS_v179)
, 0
, 0
, 0
, 0
, CONSTR(1,1,0)
, 0
, 0
, 0
, 0
, 850001
, useLabel(ST_v178)
,	/* CT_v181: (byte 0) */
  HW(1,0)
, 0
,};
Node CF_System_46Exit_46exitFailure[] = {
  VAPTAG(useLabel(FN_System_46Exit_46exitFailure))
, useLabel(PS_v177)
, 0
, 0
, 0
, VAPTAG(useLabel(FN_System_46exitWith))
,};
Node PM_System_46Exit[] = {
 	/* ST_v171: (byte 0) */
  bytes2word(83,121,115,116)
, bytes2word(101,109,46,69)
, bytes2word(120,105,116,0)
,};
Node PP_System_46Exit_46exitFailure[] = {
 };
Node PC_System_46Exit_46exitFailure[] = {
 	/* ST_v178: (byte 0) */
  bytes2word(83,121,115,116)
, bytes2word(101,109,46,69)
, bytes2word(120,105,116,46)
, bytes2word(101,120,105,116)
, bytes2word(70,97,105,108)
, bytes2word(117,114,101,0)
,};
Node PP_System_46Exit_46exitSuccess[] = {
 };
Node PC_System_46Exit_46exitSuccess[] = {
 	/* ST_v173: (byte 0) */
  bytes2word(83,121,115,116)
, bytes2word(101,109,46,69)
, bytes2word(120,105,116,46)
, bytes2word(101,120,105,116)
, bytes2word(83,117,99,99)
, bytes2word(101,115,115,0)
,	/* PS_v180: (byte 0) */
  useLabel(PM_System_46Exit)
, useLabel(PP_System_46Exit_46exitFailure)
, useLabel(PC_System_46exitWith)
,	/* PS_v179: (byte 0) */
  useLabel(PM_System_46Exit)
, useLabel(PP_System_46Exit_46exitFailure)
, useLabel(PC_System_46ExitFailure)
,	/* PS_v177: (byte 0) */
  useLabel(PM_System_46Exit)
, useLabel(PP_System_46Exit_46exitFailure)
, useLabel(PC_System_46Exit_46exitFailure)
,	/* PS_v175: (byte 0) */
  useLabel(PM_System_46Exit)
, useLabel(PP_System_46Exit_46exitSuccess)
, useLabel(PC_System_46exitWith)
,	/* PS_v174: (byte 0) */
  useLabel(PM_System_46Exit)
, useLabel(PP_System_46Exit_46exitSuccess)
, useLabel(PC_System_46ExitSuccess)
,	/* PS_v172: (byte 0) */
  useLabel(PM_System_46Exit)
, useLabel(PP_System_46Exit_46exitSuccess)
, useLabel(PC_System_46Exit_46exitSuccess)
,};
