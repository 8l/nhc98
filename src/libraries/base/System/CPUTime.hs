# 1 "System/CPUTime.hsc"
-----------------------------------------------------------------------------
# 2 "System/CPUTime.hsc"
-- |
-- Module      :  System.CPUTime
-- Copyright   :  (c) The University of Glasgow 2001
-- License     :  BSD-style (see the file libraries/base/LICENSE)
-- 
-- Maintainer  :  libraries@haskell.org
-- Stability   :  provisional
-- Portability :  portable
--
-- The standard CPUTime library.
--
-----------------------------------------------------------------------------

module System.CPUTime 
        (
         getCPUTime,       -- :: IO Integer
         cpuTimePrecision  -- :: Integer
        ) where

import Prelude

import Data.Ratio


# 28 "System/CPUTime.hsc"


# 30 "System/CPUTime.hsc"
import CPUTime ( getCPUTime, cpuTimePrecision )

# 32 "System/CPUTime.hsc"


# 67 "System/CPUTime.hsc"


# 69 "System/CPUTime.hsc"
realToInteger :: Real a => a -> Integer
realToInteger ct = round (realToFrac ct :: Double)
  -- CTime, CClock, CUShort etc are in Real but not Fractional, 
  -- so we must convert to Double before we can round it

# 74 "System/CPUTime.hsc"


# 161 "System/CPUTime.hsc"

-- |The 'cpuTimePrecision' constant is the smallest measurable difference
-- in CPU time that the implementation can record, and is given as an
-- integral number of picoseconds.


# 170 "System/CPUTime.hsc"


# 181 "System/CPUTime.hsc"
