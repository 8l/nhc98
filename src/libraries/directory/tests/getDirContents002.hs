import System.Directory

main = getDirectoryContents "nonexistent"
