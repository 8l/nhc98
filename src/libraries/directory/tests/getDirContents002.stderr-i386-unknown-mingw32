getDirContents002.exe: nonexistent: getDirectoryContents: does not exist (The system cannot find the path specified.)
