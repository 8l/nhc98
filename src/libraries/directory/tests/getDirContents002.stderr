getDirContents002: nonexistent: getDirectoryContents: does not exist (No such file or directory)
